9 sign in account

russellvaughan opened the pull request:
closes #9

russellvaughan committed:
updated readme to include user story 9, sign in, under contruction

russellvaughan committed:
WIP. added sign_in route to app.rb. added authenticate method to user model. add sign_in view. created conditional statement so no error if user nil in spaces.erb. created sign_in spec. create user model spec. removed merge conflict in spec_helper.

russellvaughan committed:
user story 9 complete
add Sinatra flash to Gemfile.
include Sinatra flash to controller. Include error message logic to sign_in and signing_up views
. amend registration form to make password required. Add sinatra flash error message display to layout form. amend sign_in to require password.

tcpickard94 commented on README.md:
````diff
@@ -35,3 +35,10 @@ As a user
+So that I can book spaces or list spaces
+I want to be able to sign-in
+```
+:construction:
````
Remove construction

tcpickard94 commented on app/app.rb:
```diff
@@ -18,10 +20,33 @@ class MakersBnb < Sinatra::Base
+
+  post '/user/sign_in' do 
+    user = User.authenticate(params[:email],params[:password])
+    if user == -1
```
Is this necessary? Could you not just have `if user  else`

jaxdid replied:
Went with a more nuanced if-else tree in order to address different errors on sign-in (non-existent user vs. incorrect password).

tcpickard94 commented on app/models/User.rb:
```diff
@@ -17,4 +18,17 @@ def password=(password)
+
+  def self.authenticate(email,password)
+  user = first(email: email)
+    if user 
```
I think this could be cleaned up to something like `user if user && BCrypt::Password.new(user.password_hash) == password`. I am not 100% sure what the 0 and -1 is for

jaxdid replied:
Kept nested if-else structure for granularity. Refactored abstract return values (`0` and `-1`) for clarity (`:incorrect_password` and `:nonexistent_user`).

tcpickard94 commented on spec/spec_helper.rb:
```diff
@@ -13,20 +13,9 @@
->>>>>>> 5c64cd8437b2995a7458f65c8cff6132c91111c7
 
   config.before(:suite) do
-    DatabaseCleaner.strategy = :transaction
```
Interested as to why you changed this

russellvaughan committed:
removed construction from README and changed datamapper cleaner strategy back to transaction

russellvaughan committed:
merge conflicts resolved

jaxdid committed:
Resolved conflicts and light refactoring

Merged master. Fixed conflicts.
Refactored User model and controller logic for clarity when checking
user login authentication.
Deleted unused `sign_out_spec.rb`.

tcpickard94 commented:
:shipit:

russellvaughan committed:
merge resolved
